packages/cli/src/error_handler.ts

```diff
@@ -5,6 +5,9 @@ import { extractSubCommands } from './extract_sub_commands.js';
 
 let hasAttachUnhandledExceptionListenersBeenCalled = false;
 
+//Create a variable to track if the error is handled
+let hasError = false;
```

Comment:
Two things.
1. The version with `Map` was better. It is possible that multiple errors arrive here from asynchronous calls during lifespan of the process.
2. Please give this symbol better name. A name is good when you don't have to write comment to explain what it does.